docs/toc/reader.rst

```diff
@@ -0,0 +1,142 @@
+Reader
+=======
+
+A pipeline component that reads data from a file path into a data iterator.
+
+Usage
+------
+
+* Read data from a path and load it into a data iterator. It returns one datum at a time.
```

Comment:
what do you mean by return a datum a time. what's the definition of a datum, and what's the definition of `a time`